File: models/branches/PharmacyPayment.js
const PharmacyPaymentSchema = new mongoose.Schema({

    hospitalId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Hospital",
      required: true
    },
  
    branchId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Branch",
      required: true
    },
  
    invoiceId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "PharmacyInvoice",
      required: true
    },
  
    paymentMode: {
      type: String,
      enum: ["Cash","Card","UPI","Insurance"]
    },
  
    amountPaid: Number,
    transactionId: String,
  
    receivedBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User"
    }
  
  }, { timestamps: true });

File: jstojv/src/main/java/com/samrat/jstojv/entity/PharmacyPayment.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "pharmacy_payments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PharmacyPayment extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "invoice_id", nullable = false)
    private PharmacyInvoice invoice;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_mode")
    private PaymentMode paymentMode;

    @Column(name = "amount_paid")
    private Double amountPaid;

    @Column(name = "transaction_id")
    private String transactionId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "received_by")
    private User receivedBy;

    public enum PaymentMode {
        Cash, Card, UPI, Insurance
    }
}
